Libs/Gtk3/Classes/Window.cs

```diff
@@ -71,18 +423,703 @@ public bool Resizable
+        }
+
+        public Widget? GetAttachedTo()
+        {
+            IntPtr widgetPtr = Native.get_attached_to(Handle);
+            return widgetPtr == IntPtr.Zero ? null : WrapPointerAs<Widget>(widgetPtr);
+        }
+
+        public Gdk.WindowTypeHint GetTypeHint() =>
+            Native.get_type_hint(Handle);
+
+        public bool GetSkipTaskbarHint() =>
+            Native.get_skip_taskbar_hint(Handle);
+
+        public bool GetSkipPagerHint() =>
+            Native.get_skip_pager_hint(Handle);
+
+        public bool GetUrgencyHint() =>
+            Native.get_urgency_hint(Handle);
+
+        public bool GetAcceptFocus() =>
+            Native.get_accept_focus(Handle);
+
+        public WindowType GetWindowType() =>
+            Native.get_window_type(Handle);
+
+        public void Move(int x, int y) =>
+            Native.move(Handle, x, y);
+
+        [Obsolete("Geometry handling in GTK is deprecated.")]
+        public bool ParseGeometry(string geometry) =>
+            Native.parse_geometry(Handle, geometry);
+
+        [Obsolete("GUI builders can call Hide(), Unrealize() and then Show() on window themselves, if they still need this functionality.")]
+        public void ReshowWithInitialSize() =>
+            Native.reshow_with_initial_size(Handle);
+
+        public void Resize(int width, int height) =>
+            Native.resize(Handle, width, height);
+
+        [Obsolete("This function does nothing. Use Window.Resize() and compute the geometry yourself.")]
+        public void ResizeToGeometry(int width, int height) =>
+            Native.resize_to_geometry(Handle, width, height);
+
+        public static void SetDefaultIconList(GLib.List list)
+        {
+            IntPtr listPtr = Marshal.AllocHGlobal(Marshal.SizeOf(list));
+            Marshal.StructureToPtr(list, listPtr, true);
+            Native.set_default_icon_list(listPtr);
+            Marshal.FreeHGlobal(listPtr);
+        }
+
+        public static void SetDefaultIcon(GdkPixbuf.Pixbuf icon) =>
+            Native.set_default_icon(GetHandle(icon));
+
+        public static bool SetDefaultIconFromFile(string filename, out GLib.Error error)
+        {
+            var result = Native.set_default_icon_from_file(filename, out IntPtr e);
+            error = Marshal.PtrToStructure<GLib.Error>(e);
+            return result;
+        }
+
+        public static void SetDefaultIconName(string name) =>
+            Native.set_default_icon_name(name);
+
+        public void SetIcon(GdkPixbuf.Pixbuf? icon) =>
+            Native.set_icon(Handle, icon is null ? IntPtr.Zero : GetHandle(icon));
+
+        public void SetIconList(GLib.List list)
+        {
+            IntPtr listPtr = Marshal.AllocHGlobal(Marshal.SizeOf(list));
+            Marshal.StructureToPtr(list, listPtr, true);
+            Native.set_icon_list(Handle, listPtr);
+            Marshal.FreeHGlobal(listPtr);
+        }
+
+        public bool SetIconFromFile(string filename, out GLib.Error error)
```

Comment:
This is probably part of a bigger discussion about whether we want to expose `GLib.Error` to the end user, or treat them as exceptions.

Currently we're using `HandleError(errPtr)` to throw a GException which the user is then responsible for handling.

See: [Object.cs#L229](https://github.com/gircore/gir.core/blob/develop/Libs/GObject/Classes/Object.cs#L229)

Until we make a decision on this, we should probably stick with `HandleError` to be consistent.

+@badcel had some thoughts on this.

Reply:
Yes we should use `HandleError` to be consistent with the rest of the code. If we decide to change error handling we can just change all the references of this method.

Reply:
I've used `HandleError` as recommended, now these functions are throwing an exception if any error occurs.